src/services/pcn-router/src/ArpTable.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 The Polycube Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO: Modify these methods with your own implementation
+
+#include "ArpTable.h"
+#include "Router.h"
+
+ArpTable::ArpTable(Router &parent, const ArpTableJsonObject &conf)
+    : ArpTableBase(parent) {
+  logger()->info("Creating ArpTable instance");
+}
+
+ArpTable::ArpTable(Router &parent, const std::string &mac,
+                   const std::string &ip, const std::string &interface)
+    : ArpTableBase(parent), mac_(mac), ip_(ip), interface_(interface) {}
+
+ArpTable::~ArpTable() {}
+
```

Comment:
Is the JSONObject implementation missing or dropped for some reason?

```	
void ArpEntry::update(const ArpEntryJsonObject &conf) {		
  // This method updates all the object/parameter in ArpEntry object specified		
  // in the conf JsonObject.		
  // You can modify this implementation.		
  setInterface(conf.getInterface());		
  setMac(conf.getMac());		
}		
ArpEntryJsonObject ArpEntry::toJsonObject() {		
  ArpEntryJsonObject conf;		
  conf.setInterface(getInterface());		
  conf.setMac(getMac());		
  conf.setAddress(getAddress());		
  return conf;		ArpTable::~ArpTable() {}
}
```

Reply:
This is part of the automatically generated code. So, it should be correct.

Reply:
In the new version of the generator it is automatically generated.
For this reason many functions including JSONObject are now found in the folder "/pcn-router/src/base" in the related files.